lib/src/test/kotlin/network/utils/NetworkConfig.kt

```diff
@@ -0,0 +1,107 @@
+package network.utils
+
+import com.swmansion.starknet.data.NetUrls
+import com.swmansion.starknet.data.types.Felt
+
+class NetworkConfig {
+    data class Config(
+        val network: Network,
+        val rpcUrl: String,
+        val gatewayUrl: String,
+        val feederGatewayUrl: String,
+        val accountAddress: Felt,
+        val privateKey: Felt,
+        val constNonceAccountAddress: Felt? = null,
+        val constNoncePrivateKey: Felt? = null,
+    ) {
+        companion object {
+
+            @JvmStatic
+            fun getDefaultGatewayURL(network: Network): String {
+                val baseUrl = when (network) {
+                    Network.INTEGRATION -> NetUrls.INTEGRATION_URL
+                    Network.TESTNET -> NetUrls.TESTNET_URL
+                }
+                return "$baseUrl/gateway"
+            }
+
+            @JvmStatic
+            fun getDefaultFeederGatewayURL(network: Network): String {
+                val baseUrl = when (network) {
+                    Network.INTEGRATION -> NetUrls.INTEGRATION_URL
+                    Network.TESTNET -> NetUrls.TESTNET_URL
+                }
+                return "$baseUrl/feeder_gateway"
+            }
+        }
+    }
+
+    enum class Network(val value: String) {
+        INTEGRATION("INTEGRATION"),
+        TESTNET("TESTNET"),
+    }
+
+    enum class NetworkTestMode {
+        DISABLED,
+        NON_GAS,
+        ALL;
+
+        companion object {
+            @JvmStatic
+            fun fromString(value: String): NetworkTestMode {
+                return values().firstOrNull { it.name.equals(value, ignoreCase = true) } ?: DISABLED
+            }
+        }
+    }
+
+    companion object {
+        private val testMode: NetworkTestMode =
+            NetworkTestMode.fromString(System.getProperty("networkTestMode") ?: "disabled")
+
+        @JvmStatic
+        fun isTestEnabled(requiresGas: Boolean): Boolean {
+            return when (testMode) {
+                NetworkTestMode.DISABLED -> false
+                NetworkTestMode.NON_GAS -> !requiresGas
+                NetworkTestMode.ALL -> true
+            }
+        }
+
+        @JvmStatic
+        private fun makeConfigFromEnv(): Config {
+            if (testMode == NetworkTestMode.DISABLED) {
+                return Config(
+                    network = Network.INTEGRATION,
+                    rpcUrl = "",
+                    gatewayUrl = "",
+                    feederGatewayUrl = "",
+                    accountAddress = Felt.ZERO,
+                    privateKey = Felt.ZERO,
+                )
+            }
+
+            val env = System.getenv()
+            val network = Network.valueOf(
+                env.getOrElse("NETWORK_TEST_NETWORK") { throw RuntimeException("NETWORK_TEST_NETWORK not found in environment variables") },
```

Comment:
I'm not a fan of `NETWORK_TEST_NETWORK` name myself  💀 
However, `NETWORK_TEST_PROFILE` seems less self-explanatory and I'm out of ideas.

Reply:
Hmm, perhaps something like `NODE_TEST_NETWORK` or even `RPCNODE_TEST_NETWORK` to further highlight that these are for real-life nodes?

BUT, I think we will be able to get rid fo this completely soon, so maybe this is not a big issue after all.

Reply:
The intention is to have a `NETWORK_` prefix for all the variables related to network tests config.
Maybe it would better to replace it with `NODE_`, but as gateway is not yet removed, this could be a bit confusing.

Reply:
that's true, but `NETWORK_TEST_NETWORK` is kind of confusing as well :D Up to you to decide really, we can always rename it after gateway is removed :+1:

Reply:
I'm thinking to go with `NETWORK_TEST_PROFILE` 🤔 
wdyt?

Reply:
A bit too vague. Maybe NETWORK_TEST_NETWORK_NAME ? or NETWORK_TEST_NETWORK_ID ?